src/App.tsx

```diff
@@ -3,33 +3,51 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string;
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  changeSearchInputHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      query: event?.target.value,
+    });
+  };
 
   render() {
+    const { query } = this.state;
+
+    const visibleMovies = [...moviesFromServer].filter(movie => {
+      return movie.title.toLowerCase().includes(query.toLowerCase())
+        || movie.description.toLowerCase().includes(query.toLowerCase());
+    });
```

Comment:
You can optimize this
```suggestion
    const visibleMovies = [...moviesFromServer].filter(({ title, description })  => {
      const lowerCaseQuery = query.toLowerCase();
      
      return title.toLowerCase().includes(lowerCaseQuery)
        || description.toLowerCase().includes(lowerCaseQuery);
    });
```

Reply:
Let's create a separate function `getVisibleMovies()` for this logic? 🙂